src/containers/QuoteCreate/QuoteCreate.tsx

```diff
@@ -45,15 +45,18 @@ class QuoteCreate extends React.Component<Props, State> {
     this.updateValue = this.updateValue.bind(this)
   }
 
-  public updateValue = (value: IQuote): void => {
+  public updateValue = (value: common.ValueType): void => {
+    if (!value || typeof value !== 'object' || Array.isArray(value)) {
+      console.warn('user input is not a JSON object')
+      return
+    }
     const quote = value
-    const result = {}
-    Object.keys(value.cost.tax).forEach(entryKey => {
+    const result: Record<string, unknown> = {}
+    Object.keys(value.cost.tax).forEach((entryKey) => {
       result[entryKey] = value.cost.tax[entryKey]
     })
     quote.cost.tax = result
-
-    this.setState({ quote })
+    this.setState({ quote: quote as IQuote })
```

Comment:
Is the quote not of type IQuote? How can we check, if it is an IQuote without forcing it?

Reply:
the json-schema editor unfortunately isn't generic, so we would have to build a custom type guard, which seems overkill for that